### ISSUE TITLE
Two swarm on same mesos cluster share jobs. Bug or Feature?

### ISSUE THREAD

**Issue by Spritekin** [docker/swarm#1895]
Hi,
I started two swarm servers in the same mesos cluster (one for tests). One was started yesterday so it has running jobs. The other was started minutes ago.

When I do a docker ps on each one, the old server lists all its jobs. But the new service, where I have not issued any job, also responds with a partial list. This is a partial extract:

Old server

```
$ docker -H tcp://172.31.46.152:2376 ps
CONTAINER ID        IMAGE                                                                   COMMAND                  CREATED             STATUS              PORTS               NAMES
8cf93e746f12        docker-registry.aws.cmcrc.com:5000/cmcrc/workflow_jobs:20160225123027   "/docker-entrypoint.s"   25 hours ago        Up 25 hours                             ip-172-31-41-173.aws.cmcrc.com/aws_depth_convert-paris_depth,ip-172-31-41-173.aws.cmcrc.com/mesos-2ce3e7f3-1712-4a4b-8338-04077c371a67-S6.f4ded419-2497-4e6a-bcfa-eddb02236dcc
752bea496733        docker-registry.aws.cmcrc.com:5000/cmcrc/workflow_jobs:20160225123027   "/docker-entrypoint.s"   25 hours ago        Up 25 hours                             ip-172-31-41-173.aws.cmcrc.com/aws_depth_convert-paris_depth,ip-172-31-41-173.aws.cmcrc.com/mesos-2ce3e7f3-1712-4a4b-8338-04077c371a67-S6.4bd6ceaa-546c-40ed-928b-194b728d3b81
```

New server:

```
$ docker -H tcp://172.31.36.151:2376 ps
CONTAINER ID        IMAGE                                                                   COMMAND                  CREATED             STATUS              PORTS               NAMES
8cf93e746f12        docker-registry.aws.cmcrc.com:5000/cmcrc/workflow_jobs:20160225123027   "/docker-entrypoint.s"   25 hours ago        Up 25 hours                             ip-172-31-41-173.aws.cmcrc.com/aws_depth_convert-paris_depth,ip-172-31-41-173.aws.cmcrc.com/mesos-2ce3e7f3-1712-4a4b-8338-04077c371a67-S6.f4ded419-2497-4e6a-bcfa-eddb02236dcc
752bea496733        docker-registry.aws.cmcrc.com:5000/cmcrc/workflow_jobs:20160225123027   "/docker-entrypoint.s"   25 hours ago        Up 25 hours                             ip-172-31-41-173.aws.cmcrc.com/aws_depth_convert-paris_depth,ip-172-31-41-173.aws.cmcrc.com/mesos-2ce3e7f3-1712-4a4b-8338-04077c371a67-S6.4bd6ceaa-546c-40ed-928b-194b728d3b81
```

On the full list (which I can't reproduce due to space) one lists 55 jobs, the other 76 jobs. So the list is not even complete. Maybe because they are running slightly different swarm versions. I would need to do more testing on this.

grodriguez:~ Guimo$ docker -H tcp://172.31.36.151:2376 ps | wc -l
      55
grodriguez:~ Guimo$ docker -H tcp://172.31.46.152:2376 ps | wc -l
      76

The point is... why does the new service is showing the jobs of the old service. Also, even when both are swarm, each one has a different service name (one is called swarm, the other is called swarmtest). I assume this is caused by the slave monitoring started internally by swarm in the code (dockerclient.go MonitorEvents).

I'm not sure if I should call this a bug or a feature. If they can really share the job list then it would mean that I can start redundant swarm servers which would be awesome.

Of course on the other side, I'm not able to start an independent service as it mixes with the other.

Luck!

**Comment by Spritekin** [comment]
I just noticed that both services can find the containers by id even if they don't list them all.

So, I just scaled my swarm service to 3 instances using marathon. WAY more stable. I can no longer detect a few bugs where a container reported as not existing. Also if one service crash then there are others to respond while it restarts.

Intentional or not, I really love they way it is working. I really don't care too much if I don't have another independent swarm manager in the cluster, this redundant service made my day.

Luck!
